storage/operation/writes_functors.go

```diff
@@ -0,0 +1,33 @@
+package operation
+
+import "github.com/onflow/flow-go/storage"
+
+// Leo: This package includes deprecated functions that wraps the operation of writing to the database.
+// They are needed because the original badger implementation is also implemented in the same wrapped function manner,
+// since badger requires writes to be done in a transaction, which is stateful.
+// Using these deprecated functions could minimize the changes during refactor and easier to review the changes.
+// The simplified implementation of the functions are in the writes.go file, which are encouraged to be used instead.
+
+func Upsert(key []byte, val interface{}) func(storage.Writer) error {
```

Comment:
Originally these functions are implemented as curried functions in order to follow the same pattern as badger transaction. 

However, since we are moving away from badger transaction, it's simpler just removing this additional wrap by using the uncurried functions. 

I extracted the uncurried functions and kept the original functions here so that the same logic are reused, as well as the tests.

Useful for refactoring storage modules with UpsertKey instead of Upsert.